antismash/common/all_orfs.py

```diff
@@ -89,26 +89,28 @@ def scan_orfs(seq: str, direction: int, offset: int = 0) -> List[FeatureLocation
     return sorted(matches, key=lambda x: min(x.start, x.end))
 
 
-def create_feature_from_location(record: Record, location: FeatureLocation,
-                                 counter: int = 1, label: Optional[str] = None) -> CDSFeature:
+def create_feature_from_location(record: Record, location: FeatureLocation, name: str) -> CDSFeature:
     """ Creates a CDS feature covering the provided location.
 
         Arguments:
             record: The Record the CDSFeature will belong to, used to generate
                     the feature translation
             location: The FeatureLocation specifying the location of the CDSFeature
-            counter: An integer to use to format a default label 'allorf' with,
-                     used only if label not provided
-            label: The locus tag, protein id, and gene name to use for the new
-                   CDSFeature
+            name: The locus tag, protein id, and gene name to use for the new
+                  CDSFeature
 
         Returns:
             The CDSFeature created.
     """
-    if label is None:
-        label = 'allorf%03d' % counter
+    if not isinstance(record, Record):
+        raise TypeError("Can not create a feature, no record to attach it to")
+    if not isinstance(location, FeatureLocation):
+        raise TypeError("Can not create a feature without location")
+    if not isinstance(name, str):
```

Comment:
This one might be better as ensuring that the string isn't empty
```suggestion
    if not name or not isinstance(name, str):
        raise ValueError("name must be a non-empty string")
```